framework/include/userobject/InterfaceQpMaterialPropertyBaseUserObject.h

```diff
@@ -0,0 +1,113 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "InterfaceQpUserObjectBase.h"
+
+// Forward declarations
+template <typename T = Real>
+class InterfaceQpMaterialPropertyBaseUserObject;
+
+template <>
+InputParameters validParams<InterfaceQpMaterialPropertyBaseUserObject<>>();
```

Comment:
@permcody can correct me, but since this way of doing `validParams` is "legacy" we probably shouldn't add it for new objects

Reply:
As I explained below, I tried, but I wasn't able to remove the line. @permcody do you have any example of registered class without the incremented line

Reply:
It's pretty lame if this is simultaneously legacy and we can't create new objects without it

Reply:
@lindsayad for some of the objects I can remove the valiParam 
for other I get this kid of error but I don't understand why
```
/home/arovinelli/projects/NEW_stuff/moose/framework/src/userobject/InterfaceValueUserObject.C:15:39: error: specialization of ‘InputParameters validParams() [with T = InterfaceValueUserObject]’ after instantiation
 validParams<InterfaceValueUserObject>()
```

the incriminated classes are `InterfaceValueUserObject` and `InterfaceQpUserObjectBase`

Reply:
This was my fault, a template was missing a `T` in a templete